### ISSUE TITLE
bug: Error from fetching event logs

### ISSUE THREAD

**Issue by webelf101** [palomachain/paloma#752]
# What is happening?
When the block time is short and there isn't valsetupdate for a long time, pigeon requests fetching logs from rpc service api and then it fails as too many requests.

# How to solve?
1. call `isArbitraryCallAlreadyExecuted()` function only once with maximum fetch block length(9999).
2. modify findLastValsetMessageID() function so that it don't fetch log but read last valset id from the compass-evm smart contract on blockchain.
3. and update compass-evm smart contract so that it can return the last valset id.

**Comment by webelf101** [comment]
#1 is already done on my side.

**Comment by taariq** [comment]
@webelf101 please include the files where the functions ae called. I'll assign to @mszuyev

**Comment by taariq** [comment]
@webelf101 also, please close and open new issue on https://github.com/VolumeFi/paloma/issues
